include/sys/dsl_scan.h

```diff
@@ -76,6 +79,19 @@ typedef enum dsl_scan_flags {
 
 #define	DSL_SCAN_FLAGS_MASK (DSF_VISIT_DS_AGAIN)
 
+typedef struct dsl_rescrub_phys {
+	uint64_t rescrub_func; /* pool_scan_func_t */
+	uint64_t rescrub_state; /* dsl_scan_state_t */
```

Comment:
The naming convention for struct prefixes here would probably be something like `drp_state`, but I'm open to discussion on whether that's the right move.